src/twisted/internet/test/test_base.py

```diff
@@ -155,6 +158,32 @@ def fakeGetHostByName(name):
         result.put(IOError("The I/O was errorful"))
 
 
+    def test_resolverGivenBytes(self):
+        """
+        L{ThreadedResolver.getHostByName} is passed L{bytes}.
+        """
+        calls = []
+
+        @implementer(IResolverSimple)
+        class FakeResolver(object):
+            def getHostByName(self, name, timeouts=()):
+                calls.append(name)
+                return Deferred()
+
+        class JustEnoughReactor(ReactorBase):
+            def installWaker(self):
+                pass
+
+        fake = FakeResolver()
+        reactor = JustEnoughReactor()
+        reactor.installResolver(fake)
+        rec = FirstOneWins(Deferred())
+        reactor.nameResolver.resolveHostName(rec, u"xn--pxaix.localhost")
```

Comment:
1. please use `.invalid` rather than `.localhost`, or `something.example.com`, RFCs are great
2. please make the test a bit more rigorous in terms of `resolveHostName` being passed *both* the ascii-as-unicode given here *and* the actual cyrillic text, making sure that the underlying `FakeResolver` is passed the same value in both cases.